manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -235,6 +235,9 @@ data:
         metadata:
           labels:
             security.istio.io/tlsMode: {{ index .ObjectMeta.Labels `security.istio.io/tlsMode` | default "istio"  | quote }}
+            {{- if eq (index .ProxyConfig.ProxyMetadata "ISTIO_META_ENABLE_HBONE") "true" }}
```

Comment:
I really dislike the term HBONE. What this really means is accepting mTLS HTTP CONNECT (only)? Why restrict to h2? h2 is objectively worse than HTTP/1.1 for some cases.

Reply:
If you'd like to change the protocol we use I suggest proposing in another design doc. I don't necessarily disagree with you, but this PR is implementing the design which has been established for ~3 years and gone through plenty of discussion and naming bike shedding; this PR isn't a good place to try to change that.

Reply:
Yes, I'm not calling to change the PR and design docs. This is  a purely internal annotation. HBONE doesn't tell anything about HTTP/TCP/UDP etc.

Reply:
And it shouldn't tell that - there are well defined ways to identify h2 vs h1 ( ALPN ) and include metadata (headers, labels). 

What I would like is an option (env variable?) to skip the auto-hbone and just do exclusive hbone and nothing else.